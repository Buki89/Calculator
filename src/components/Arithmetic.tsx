import React from "react";
import Number from "./Number";
import styled from "styled-components";

const Box = styled.div`
  margin: 30px;
  width: 100px;
  display: flex;
  flex-direction: column;
`;

interface Props {
  handleClick: any;
  handleMakeResult: any;
}

const Arithmetic = (props: Props) => {
  const { handleClick, handleMakeResult } = props;

  return (
    <Box>
      <Number handleClick={handleClick} numbers={["+", "-", "*", "/"]} />
      <Number handleClick={handleMakeResult} numbers={["="]} />
    </Box>
  );
};

export default Arithmetic;
