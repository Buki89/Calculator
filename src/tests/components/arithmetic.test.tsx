import React from "react";
import renderer from "react-test-renderer";
import Arithmetic from "../../components/Arithmetic";

it("renders correctly when there are no items", () => {
  const tree = renderer
    .create(<Arithmetic handleClick handleMakeResult />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
