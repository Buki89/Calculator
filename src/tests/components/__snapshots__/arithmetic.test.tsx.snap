// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="sc-gtssRu eSBDSp"
>
  <div
    className="sc-bdnylx kxBuNG"
    onClick={[Function]}
  >
    +
  </div>
  <div
    className="sc-bdnylx kxBuNG"
    onClick={[Function]}
  >
    -
  </div>
  <div
    className="sc-bdnylx kxBuNG"
    onClick={[Function]}
  >
    *
  </div>
  <div
    className="sc-bdnylx kxBuNG"
    onClick={[Function]}
  >
    /
  </div>
  <div
    className="sc-bdnylx kxBuNG"
    onClick={[Function]}
  >
    =
  </div>
</div>
`;
